synapse/lib/stormlib/gen.py

```diff
@@ -425,6 +453,53 @@ class LibGen(s_stormtypes.Lib):
             [ geo:place=(gen, name, $geoname) :name=$geoname ]
             return($node)
         }
+
+        function fileBytesBySha256(sha256, try=$lib.false) {
+            ($ok, $sha256) = $__maybeCast($try, hash:sha256, $sha256)
+            if (not $ok) { return() }
+
+            file:bytes:sha256=$sha256
+            return($node)
+
+            [ file:bytes=$sha256 ]
+            return($node)
+        }
+
+        function cryptoX509CertBySha256(sha256, try=$lib.false) {
+            ($ok, $sha256) = $__maybeCast($try, hash:sha256, $sha256)
+            if (not $ok) { return() }
+
+            $guid = $lib.guid(valu=$sha256)
+
+            // Try to lift crypto:x509:cert by guid
+            crypto:x509:cert=$guid
+            return($node)
+
+            // Try to lift crypto:x509:cert by sha256
+            crypto:x509:cert:sha256=$sha256
+            return($node)
+
+            // Try to lift crypto:x509:cert by file
+            file:bytes:sha256=$sha256 -> crypto:x509:cert:file
+            { -:sha256 [ :sha256 = $sha256 ] }
+            return($node)
+
+            // Create a new crypto:x509:cert with file and sha256
+            [ crypto:x509:cert=$guid
+                :file = $fileBytesBySha256($sha256, try=$try)
```

Comment:
Minor, but we shouldn't have to try here since we already normed the sha256 right?